### PR TITLE
Use the default number of CPUs

### PR DESCRIPTION
**Pull request by tamird**
This reverts commit 30b6050562a8f4298a5b9c4aeeea9a964bb60ac4.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/1839)

### CODE REVIEW

File: gossip/client_test.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func TestClientGossip(t *testing.T) {
 	rpcContext := rpc.NewContext(&base.Context{Insecure: true}, lclock, stopper)
 	client.start(local, disconnected, rpcContext, stopper)
 
-	util.SucceedsWithin(t, 500*time.Millisecond, func() error {
+	util.SucceedsWithin(t, 5*time.Minute, func() error {
```
Comment (tschottdorf) [inline gossip/client_test.go]: why that high?

Comment (tamird) [inline gossip/client_test.go]: i was debugging, see second commit

**Comment by tamird** [comment]
OK, I think this once again ready for prime time. There are 5 tests being suppressed here.
I've run 10 builds with this in CI ([results](https://circleci.com/gh/cockroachdb/cockroach/tree/pull%2F1839)):
- 6 successes
- 2 `TestTxnDBLostUpdateAnomaly` failure (these tests are somewhat flaky by their nature)
- 1 `Too many goroutines running after tests` failure in `cli`; looks like slow or bad shutdown
- 1 `TestRaftLogQueue` failure (looks like it just took too long. can't repro locally after 4000 runs)

I think this is about as good as we're gonna get, assuming we're ok with the 5 suppressions.

@tschottdorf @bdarnell could you both please weigh in? everyone else is welcome to comment as well, of course.

**Comment by tschottdorf** [comment]
I think we should merge them. Ultimately the "big ones" here have @bdarnell's name on it so he should weigh in, but according to @tamird all test errors are reproducible locally (with sufficient iteration), so we're not losing much by skipping the tests in CI as long as we prioritize the corresponding issues accordingly. Plus, the introduction of new flaky tests has pretty much kept up with the rate at which we can fix them, and @tamird has sysiphus'ed it for long enough.

**Comment by bdarnell** [comment]
I agree we should go ahead and merge this even if we have to skip some tests, but four failures out of ten is still unacceptably high. Looks like TestTxnDBLostUpdateAnomaly should also be skipped, and maybe something in `cli`. I'd like to see this branch pass 9 of its last 10 builds before merging, even if that means expanding the list of skipped tests.

**Comment by tamird** [comment]
I don't think these are representative results, so I'm running 10 more builds (prior to this batch of 10, I had hardly seen `TestTxnDBLostUpdateAnomaly` fail, and had never seen the other two failures).

**Comment by tamird** [comment]
OK, 6 of those 10 failed:
- 1 https://github.com/cockroachdb/cockroach/issues/3025
- 1 legit new race in gossip, fixed here
- 4 other failures, all of which are now skipped (2 of these in the same run)
- 1 [timeout](https://circleci.com/gh/cockroachdb/cockroach/9328) on `Stopper.Quiesce` in `TestLogic`, when `TestLogic` was panicking (after a call to `t.Fatalf`). This one is fixed by re-panicking in `Stopper.Stop`. The underlying `Fatalf` was unfortunately not printed.

10 more builds are now running.

**Comment by tamird** [comment]
10 for 10, merging

**Pull request merged**